raja@0.14.0 +rocm ^blt@0.5.0 build fails: cmake: get_target_property() called with non-existent target "hip_runtime"

### Steps to reproduce the issue

`raja@0.14.0 +rocm ^blt@0.5.0` build fails using:
* `spack@develop` (a9996db35c1a018c582a67af112cc9f8f72e477f from `Tue Mar 8 21:22:40 2022 +0100`)
* Ubuntu 20.04, X86_64
* GCC 9.3.0
* ROCm 4.5.2
* `^blt@0.5.0`

**Concrete spec**
[raja.spec.json.txt](https://github.com/spack/spack/files/8211144/raja.spec.json.txt)

**Concretization**
```
raja@0.14.0%gcc@9.3.0~cuda+examples+exercises~ipo~openmp+rocm+shared~tests amdgpu_target=gfx908 build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
    ^blt@0.5.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
        ^cmake@3.20.6%gcc@9.3.0~doc+ncurses+openssl+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-x86_64
            ^ncurses@6.2%gcc@9.3.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-x86_64
                ^pkgconf@1.8.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
            ^openssl@1.1.1m%gcc@9.3.0~docs certs=system arch=linux-ubuntu20.04-x86_64
                ^perl@5.34.0%gcc@9.3.0+cpanm+shared+threads arch=linux-ubuntu20.04-x86_64
                    ^berkeley-db@18.1.40%gcc@9.3.0+cxx~docs+stl patches=b231fcc arch=linux-ubuntu20.04-x86_64
                    ^bzip2@1.0.8%gcc@9.3.0~debug~pic+shared arch=linux-ubuntu20.04-x86_64
                        ^diffutils@3.8%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                            ^libiconv@1.16%gcc@9.3.0 libs=shared,static arch=linux-ubuntu20.04-x86_64
                    ^gdbm@1.19%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^readline@8.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^zlib@1.2.11%gcc@9.3.0+optimize+pic+shared arch=linux-ubuntu20.04-x86_64
    ^camp@0.2.2%gcc@9.3.0~cuda~ipo+rocm~tests amdgpu_target=gfx908 build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
        ^hip@4.5.2%gcc@9.3.0~ipo build_type=Release patches=24ca25a,2a41904 arch=linux-ubuntu20.04-x86_64
            ^comgr@4.5.2%gcc@9.3.0~ipo build_type=Release arch=linux-ubuntu20.04-x86_64
                ^llvm-amdgpu@4.5.2%gcc@9.3.0~ipo~link_llvm_dylib~llvm_dylib+openmp+rocm-device-libs build_type=Release patches=d999f3b arch=linux-ubuntu20.04-x86_64
                    ^hwloc@2.7.0%gcc@9.3.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~opencl+pci~rocm+shared arch=linux-ubuntu20.04-x86_64
                        ^libpciaccess@0.16%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                            ^libtool@2.4.6%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                                ^m4@1.4.19%gcc@9.3.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
                                    ^libsigsegv@2.13%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                            ^util-macros@1.19.3%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^libxml2@2.9.12%gcc@9.3.0~python arch=linux-ubuntu20.04-x86_64
                            ^xz@5.2.5%gcc@9.3.0+pic libs=shared,static arch=linux-ubuntu20.04-x86_64
                    ^libelf@0.8.13%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^perl-data-dumper@2.173%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^python@3.8.12%gcc@9.3.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-ubuntu20.04-x86_64
                        ^expat@2.4.6%gcc@9.3.0+libbsd arch=linux-ubuntu20.04-x86_64
                            ^libbsd@0.11.5%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                                ^libmd@1.0.4%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^gettext@0.21%gcc@9.3.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu20.04-x86_64
                            ^tar@1.34%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^libffi@3.4.2%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^sqlite@3.37.2%gcc@9.3.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-ubuntu20.04-x86_64
                        ^util-linux-uuid@2.37.4%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^z3@4.8.14%gcc@9.3.0~gmp~ipo~python build_type=RelWithDebInfo arch=linux-ubuntu20.04-x86_64
                ^rocm-cmake@4.5.2%gcc@9.3.0~ipo+ldconfig build_type=Release arch=linux-ubuntu20.04-x86_64
            ^hsa-rocr-dev@4.5.2%gcc@9.3.0+image~ipo+shared build_type=Release patches=71e6851 arch=linux-ubuntu20.04-x86_64
                ^hsakmt-roct@4.5.2%gcc@9.3.0~ipo+shared build_type=Release patches=f926273 arch=linux-ubuntu20.04-x86_64
                    ^libdrm@2.4.110%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^docbook-xml@4.5%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^docbook-xsl@1.79.2%gcc@9.3.0 patches=a92c397 arch=linux-ubuntu20.04-x86_64
                        ^libpthread-stubs@0.4%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^meson@0.61.2%gcc@9.3.0 patches=aa6c50d arch=linux-ubuntu20.04-x86_64
                            ^ninja@1.10.2%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                            ^py-pip@21.3.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                            ^py-setuptools@59.4.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                                ^py-wheel@0.37.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^numactl@2.0.14%gcc@9.3.0 patches=4e1d78c,62fc8a8,ff37630 arch=linux-ubuntu20.04-x86_64
                        ^autoconf@2.69%gcc@9.3.0 patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
                        ^automake@1.16.5%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^xxd-standalone@8.2.1201%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
            ^mesa@21.3.1%gcc@9.3.0+glx~llvm+opengl~opengles+osmesa~strip buildtype=debugoptimized default_library=shared patches=ada85be arch=linux-ubuntu20.04-x86_64
                ^bison@3.8.2%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^flex@2.6.3%gcc@9.3.0+lex~nls arch=linux-ubuntu20.04-x86_64
                    ^findutils@4.8.0%gcc@9.3.0 patches=440b954 arch=linux-ubuntu20.04-x86_64
                ^glproto@1.4.17%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^libunwind@1.5.0%gcc@9.3.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs+pic+tests+weak_backtrace+xz~zlib components=none libs=shared,static arch=linux-ubuntu20.04-x86_64
                ^libx11@1.7.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^inputproto@2.3.2%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^kbproto@1.0.7%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^libxcb@1.14%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^libxau@1.0.8%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                            ^xproto@7.0.31%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^libxdmcp@1.1.2%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^xcb-proto@1.14.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^xextproto@7.3.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^xtrans@1.3.5%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^libxext@1.3.3%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^libxt@1.1.5%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^libice@1.0.9%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^libsm@1.2.3%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^py-mako@1.1.6%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^py-markupsafe@2.0.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                ^xrandr@1.5.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                    ^libxrandr@1.5.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^libxrender@0.9.10%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                            ^renderproto@0.11.1%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
                        ^randrproto@1.5.0%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
            ^rocminfo@4.5.2%gcc@9.3.0~ipo build_type=Release arch=linux-ubuntu20.04-x86_64
            ^roctracer-dev-api@4.5.2%gcc@9.3.0 arch=linux-ubuntu20.04-x86_64
```

**Error**
Reproduced using Docker container image `ecpe4s/ubuntu20.04-runner-x86_64:2022-02-01`
```
$> spack mirror add E4S https://cache.e4s.io
$> spack buildcache keys -it

$> spack install --cache-only --include-build-deps --only dependencies -f ./raja.spec.json

$> spack install --no-cache --only package -f ./raja.spec.json
...
==> Installing raja-0.14.0-n56ixwxkndflxwwngog4pzh2dstzqddh
==> No patches needed for raja
==> raja: Executing phase: 'initconfig'
==> raja: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/raja-0.14.0-n56ixwxkndflxwwngog4pzh2dstzqddh' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/raja-0.14.0-n56ixwxkndflxwwngog4pzh2dstzqddh/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/raja-0.14.0-n56ixwxkndflxwwngog4pzh2dstzqddh/lib64;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/camp-0.2.2-jipxbimalfjb4h4p3wgdb33v44asnpzf/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hip-4.5.2-acjk4zgstai6dqzd53kwcvhttllzrese/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/comgr-4.5.2-vform5q7kdm7qlbymgpjo2rwpysytbrx/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/llvm-amdgpu-4.5.2-qb4wlxghtnvdhop7xeyt5657gkbe4gpj/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hwloc-2.7.0-hswya4sr4io27hwpjsb7oyhgqb6vzy34/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-4jbn7mznw664dc4mlseouzovsfx6sor5/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxml2-2.9.12-xlbi5rk675mptoyjnuff4q3e2lajs2wr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libiconv-1.16-zth4ohvj7y3p532iqjw7fptg3kkn3ixv/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xz-5.2.5-mq4ikt5q4zttamuxqkqe66jlftjbvkf4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/zlib-1.2.11-ufqpqnpj5gjkfx5sd2o3twwveuowcz2b/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/ncurses-6.2-nxjjphalwbf2ywlxpn74nsfcsrq2bbtd/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libelf-0.8.13-o53xuefowaaefem7z2zfar2ftwgyyezy/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/z3-4.8.14-bvcp2nq4v7f22lbflvor2m6s7wrsymjp/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hsa-rocr-dev-4.5.2-khoys46lhbnchyzknzpmeqcyxidv76cd/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hsakmt-roct-4.5.2-jtao2ghxkslpbh3klzdtzr4tsovh5nas/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libdrm-2.4.110-qnu5fdbtpnm5phkaolwe3thvdq5yarca/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpthread-stubs-0.4-wngdhpdeuvwyfid6b7rkvbhbm22e4dzr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/numactl-2.0.14-e3evcwcbrfcfnv4j2h55y6n6dciu37wi/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mesa-21.3.1-2a3nphscjfzczreiqn56t6sxhpis2p4o/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/expat-2.4.6-xbiewd3hygm2fepoiowzdtgqhyslot6e/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libbsd-0.11.5-4hjkhnqyexdjto6krttmpd5nel4geaxk/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libmd-1.0.4-bumbhuokazj5omddfnsnb4kcgfnimnnu/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/glproto-1.4.17-sphouu7fembhysc55gv7oovholhmuqcs/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libunwind-1.5.0-ahl4twb2asjfxmxeol562w4ggjngurc6/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libx11-1.7.0-dmo574l3sty7v4ssdvdbpqwasoxvgsxe/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/inputproto-2.3.2-xbobkehjb4sfrer4oveumgxzln552jgd/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/kbproto-1.0.7-xyjeeo2b6j5m2c3xfi5qvv7alxmci3qh/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxcb-1.14-5jqj6vrxbrnonljiz4tknbzjvytrg3yr/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxau-1.0.8-2f3vsx23iochykfdq57nbhzaq6zspipj/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xproto-7.0.31-kyzpwipokb7xsz5f4x4io5tim4p763fp/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxdmcp-1.1.2-obf4cvqxobn4jecysepumqjdie76iuod/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xcb-proto-1.14.1-cimfugd3wmc7sw2wkk2v2fghabo4apg3/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xextproto-7.3.0-2oagha2n6vf6wmh5lz4o4wxw6zuba6hc/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxext-1.3.3-nvxfsdo5jwr6vox6ksxiq74emjjtnno4/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxt-1.1.5-wazssbkxhwkqk4cuqmhx2i7xt4f5a7fw/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libice-1.0.9-vnalrzdkzwi2yrfj3d7qaaxsmrag7pnq/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libsm-1.2.3-qbjfft7t2dfnmwoa4yldameatse7bpoa/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/util-linux-uuid-2.37.4-gddk6t46sxk4ekce3hwmndgdxwm7fd7b/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxrandr-1.5.0-bdxvyvwxt42lxjtmzyvop4ydwlmokilm/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxrender-0.9.10-zatujbqwf2pkg5iktxowozvezgrfqeth/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/renderproto-0.11.1-uugdb56dvocbwsmnizp4hksiop22uec5/lib;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/randrproto-1.5.0-6pobkl3pic2kb3oanmdpxkz55wboobiw/lib' '-DCMAKE_PREFIX_PATH:STRING=/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/camp-0.2.2-jipxbimalfjb4h4p3wgdb33v44asnpzf;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hip-4.5.2-acjk4zgstai6dqzd53kwcvhttllzrese;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/roctracer-dev-api-4.5.2-5xq46swsc4x7x4q5wjtpxkhmgi5nkuk2;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/rocminfo-4.5.2-2wwv54tez2w2crnaufvbe7khmiyeydpi;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/mesa-21.3.1-2a3nphscjfzczreiqn56t6sxhpis2p4o;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xrandr-1.5.0-sy5agznsunkhb54z7mt2hp4l4iryv4os;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxrandr-1.5.0-bdxvyvwxt42lxjtmzyvop4ydwlmokilm;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/randrproto-1.5.0-6pobkl3pic2kb3oanmdpxkz55wboobiw;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxrender-0.9.10-zatujbqwf2pkg5iktxowozvezgrfqeth;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/renderproto-0.11.1-uugdb56dvocbwsmnizp4hksiop22uec5;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxt-1.1.5-wazssbkxhwkqk4cuqmhx2i7xt4f5a7fw;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libsm-1.2.3-qbjfft7t2dfnmwoa4yldameatse7bpoa;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libice-1.0.9-vnalrzdkzwi2yrfj3d7qaaxsmrag7pnq;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxext-1.3.3-nvxfsdo5jwr6vox6ksxiq74emjjtnno4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libx11-1.7.0-dmo574l3sty7v4ssdvdbpqwasoxvgsxe;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xtrans-1.3.5-ve4eglxuoxe7nsdcyiy4lbcvj2qegxex;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xextproto-7.3.0-2oagha2n6vf6wmh5lz4o4wxw6zuba6hc;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxcb-1.14-5jqj6vrxbrnonljiz4tknbzjvytrg3yr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xcb-proto-1.14.1-cimfugd3wmc7sw2wkk2v2fghabo4apg3;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxdmcp-1.1.2-obf4cvqxobn4jecysepumqjdie76iuod;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxau-1.0.8-2f3vsx23iochykfdq57nbhzaq6zspipj;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xproto-7.0.31-kyzpwipokb7xsz5f4x4io5tim4p763fp;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/kbproto-1.0.7-xyjeeo2b6j5m2c3xfi5qvv7alxmci3qh;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/inputproto-2.3.2-xbobkehjb4sfrer4oveumgxzln552jgd;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libunwind-1.5.0-ahl4twb2asjfxmxeol562w4ggjngurc6;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/glproto-1.4.17-sphouu7fembhysc55gv7oovholhmuqcs;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hsa-rocr-dev-4.5.2-khoys46lhbnchyzknzpmeqcyxidv76cd;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hsakmt-roct-4.5.2-jtao2ghxkslpbh3klzdtzr4tsovh5nas;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/numactl-2.0.14-e3evcwcbrfcfnv4j2h55y6n6dciu37wi;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libdrm-2.4.110-qnu5fdbtpnm5phkaolwe3thvdq5yarca;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpthread-stubs-0.4-wngdhpdeuvwyfid6b7rkvbhbm22e4dzr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/comgr-4.5.2-vform5q7kdm7qlbymgpjo2rwpysytbrx;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/llvm-amdgpu-4.5.2-qb4wlxghtnvdhop7xeyt5657gkbe4gpj;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/z3-4.8.14-bvcp2nq4v7f22lbflvor2m6s7wrsymjp;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/util-linux-uuid-2.37.4-gddk6t46sxk4ekce3hwmndgdxwm7fd7b;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/expat-2.4.6-xbiewd3hygm2fepoiowzdtgqhyslot6e;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libbsd-0.11.5-4hjkhnqyexdjto6krttmpd5nel4geaxk;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libmd-1.0.4-bumbhuokazj5omddfnsnb4kcgfnimnnu;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libelf-0.8.13-o53xuefowaaefem7z2zfar2ftwgyyezy;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/hwloc-2.7.0-hswya4sr4io27hwpjsb7oyhgqb6vzy34;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libxml2-2.9.12-xlbi5rk675mptoyjnuff4q3e2lajs2wr;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/xz-5.2.5-mq4ikt5q4zttamuxqkqe66jlftjbvkf4;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libpciaccess-0.16-4jbn7mznw664dc4mlseouzovsfx6sor5;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/blt-0.5.0-cvlyjrg5gqpmcrgwaghzy5qxzko56woy;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/cmake-3.20.6-ju2fnpe6suea7wsih7fwlwd6edh6mjz6;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/zlib-1.2.11-ufqpqnpj5gjkfx5sd2o3twwveuowcz2b;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/libiconv-1.16-zth4ohvj7y3p532iqjw7fptg3kkn3ixv;/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-9.3.0/ncurses-6.2-nxjjphalwbf2ywlxpn74nsfcsrq2bbtd' '-C' '/tmp/root/spack-stage/spack-stage-raja-0.14.0-n56ixwxkndflxwwngog4pzh2dstzqddh/spack-src/97ac826def28-linux-ubuntu20.04-x86_64-gcc@9.3.0.cmake' '/tmp/root/spack-stage/spack-stage-raja-0.14.0-n56ixwxkndflxwwngog4pzh2dstzqddh/spack-src'

2 errors found in build log:
     78      cmake/SetupPackages.cmake:63 (find_package)
     79      CMakeLists.txt:172 (include)
     80    This warning is for project developers.  Use -Wno-dev to suppress it.
     81
     82    -- Could NOT find ROCPRIM (missing: ROCPRIM_INCLUDE_DIRS)
     83    -- Using RAJA rocPRIM submodule.
  >> 84    CMake Error at cmake/SetupPackages.cmake:89 (get_target_property):
     85      get_target_property() called with non-existent target "hip_runtime".
     86    Call Stack (most recent call first):
     87      CMakeLists.txt:172 (include)
     88
     89
  >> 90    CMake Error at CMakeLists.txt:214 (message):
     91      Trying to use HIP/ROCm version .  RAJA requires HIP/ROCm version 3.5 or
     92      newer.
     93
     94
     95    -- Configuring incomplete, errors occurred!
     96    See also "/tmp/root/spack-stage/spack-stage-raja-0.14.0-n56ixwxkndflxwwngog4pzh2dstzqddh/spack-build-n56ixwx/CMakeFiles/CMakeOutput.log".
```

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.17.1-1497-a9996db35c
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-02-cmake-out.txt](https://github.com/spack/spack/files/8211151/spack-build-02-cmake-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/8211152/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/8211153/spack-build-out.txt)

@davidbeckingsale @white238 @wspear 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/raja/package.py
+++ b/var/spack/repos/builtin/packages/raja/package.py
@@ -51,7 +51,8 @@ class Raja(CachedCMakePackage, CudaPackage, ROCmPackage):
     variant('tests', default=False, description='Build tests')
 
     depends_on('blt')
-    depends_on('blt@0.4.1:', type='build', when='@0.14.0:')
+    depends_on('blt@0.5.0:', type='build', when='@0.14.1:')
+    depends_on('blt@0.4.1', type='build', when='@0.14.0')
     depends_on('blt@0.4.0:', type='build', when='@0.13.0')
     depends_on('blt@0.3.6:', type='build', when='@:0.12.0')
 

--- a/var/spack/repos/builtin/packages/umpire/package.py
+++ b/var/spack/repos/builtin/packages/umpire/package.py
@@ -75,7 +75,8 @@ class Umpire(CachedCMakePackage, CudaPackage, ROCmPackage):
     depends_on('cmake@3.9:', when='+cuda', type='build')
     depends_on('cmake@:3.20', when='+rocm', type='build')
 
-    depends_on('blt@0.4.1:', type='build', when='@6.0.0:')
+    depends_on('blt@0.5.0:', type='build', when='@6.0.1:')
+    depends_on('blt@0.4.1', type='build', when='@6.0.0')
     depends_on('blt@0.4.0:', type='build', when='@4.1.3:5.0.1')
     depends_on('blt@0.3.6:', type='build', when='@:4.1.2')
 

